es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/factory/lfactory.clj

```diff
@@ -0,0 +1,94 @@
+(ns cmr.elasticsearch.plugins.spatial.factory.lfactory
+  (:import
+   (cmr.elasticsearch.plugins SpatialScript)
+   (java.util Map)
+   (org.apache.lucene.index LeafReaderContext)
+   (org.elasticsearch.common.settings Settings)
+   (org.elasticsearch.common.xcontent.support XContentMapValues)
+   (org.elasticsearch.search.lookup SearchLookup))
+  (:require
+   [clojure.string :as str]
+   [cmr.spatial.derived :as d]
+   [cmr.spatial.point :as point]
+   [cmr.spatial.polygon :as poly]
+   [cmr.spatial.relations :as relations]
+   [cmr.spatial.serialize :as srl])
+  (:gen-class
+   :name cmr.elasticsearch.plugins.SpatialScriptLeafFactory
+   :implements [org.elasticsearch.script.FilterScript$LeafFactory]
+   :constructors {[java.util.Map org.elasticsearch.search.lookup.SearchLookup] []}
+   :init init
+   :state data))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;                    Begin factory helper functions                         ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(def parameters
+  "The parameters to the Spatial script"
+  [:ords :ords-info])
+
+(defn- extract-params
+  "Extracts the parameters from the params map given in the script."
+  [script-params]
+  (when script-params
+    (into {} (for [param parameters]
+               [param (XContentMapValues/nodeStringValue
+                        (get script-params (name param)) nil)]))))
+
+(defn- assert-required-parameters
+  "Asserts that all the parameters are supplied or it throws an exception."
+  [params]
+  (when-not (every? params parameters)
+    (throw (IllegalArgumentException.
+             (str "Missing one or more of required parameters: "
+                  (clojure.string/join parameters ", "))))))
+
+(defn- convert-params
+  "Convert the comma separated string into a vector of integers."
+  [[k v]]
+  [k (map #(Integer. ^String %) (str/split v #","))])
```

Comment:
We will get some exceptions here, do we validate the params well enough to not care?